pkg/option/config.go

```diff
@@ -3036,8 +3030,8 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 	c.AutoCreateCiliumNodeResource = vp.GetBool(AutoCreateCiliumNodeResource)
 	c.BPFRoot = vp.GetString(BPFRoot)
 	c.CGroupRoot = vp.GetString(CGroupRoot)
-	c.ClusterID = vp.GetUint32(ClusterIDName)
-	c.ClusterName = vp.GetString(ClusterName)
+	c.ClusterID = vp.GetUint32(clustermeshTypes.OptClusterID)
```

Comment:
Could we use the new type here ie c.ClusterIDName = NewClusterIDName(...)?

Reply:
This is tricky, because the only way to keep the current names of the fields (to avoid codebase-wide changes) would be to embed the `ClusterInfo` struct into `DaemonConfig`, which feels weird as it would also inherit the associated methods. Although suboptimal, I'd keep the current approach.

Reply:
Thanks Marco! I see what you are saying. Figured it might be an opportunity to begin cleaning this insanely large struct in favor of segmenting based on config domain. Using named association would scope the delegation vs `inheriting` the behaviors...  But in this case clusterInfo validation when it comes to configuration would be a good thing anyway. But we would need to establish pattens across the board which is a big undertaking at this stage ;(

Reply:
I definitely second the idea of removing configurations from the DaemonConfig struct. I took another pass at removing a few more usages of the ClusterID/ClusterName fields when possible (mainly the ones which would also be useful in the context of https://github.com/cilium/cilium/pull/27520, but I wouldn't go further as not to excessively increase the scope of this PR.

Reply:
How about instead of pulling them from viper like this, we would instead fill them into `DaemonConfig` from an `Invoke` that is part of the config cell (see my earlier comment)? Or alternatively leave it as-is, if these are going away soon anyway.

Reply:
The main issue of pulling it from an invoke hook is that we would postpone a bit the initialization, which would be an issue if any (legacy) component not directly depending on it would try to read these values. The current solution seems a bit less error prone to me from that point of view.